Zend/zend_language_parser.y

```diff
@@ -1324,6 +1330,8 @@ function_call:
 			{ $$ = zend_ast_create(ZEND_AST_STATIC_CALL, $1, $3, $4); }
 	|	variable_class_name T_PAAMAYIM_NEKUDOTAYIM member_name argument_list
 			{ $$ = zend_ast_create(ZEND_AST_STATIC_CALL, $1, $3, $4); }
+	|	new_dereferenceable T_PAAMAYIM_NEKUDOTAYIM member_name argument_list
```

Comment:
I don't think you need to list new_dereferenceable in quite this many places -- just having it in fully_dereferenceable and callable_expr should be enough. Something along these lines seems to work: https://gist.github.com/nikic/375e4a3a0c0adb7911ac75200c8f45c7

Reply:
Yes, thank you, that's much cleaner.